### ISSUE TITLE
[Feature] Add Proxy Support

### ISSUE THREAD

**Issue by ghbjklhv** [ipfs/go-ipfs#6427]
**Feature**: Add [Proxy](https://www.wikipedia.org/wiki/Proxy_server) Support
**Why?**: Adding proxy support may enable users to route connections through another server  to provide better anonymity. Plus, it would enable users to utilize IPFS with [TOR](https://www.wikipedia.org/wiki/Tor_(anonymity_network)).

_________________

Want to support this issue? Post a [bounty on it](https://www.bountysource.com/issues/75310553-feature-add-proxy-support)! We accept bounties via [BountySource](https://www.wikipedia.org/wiki/Bountysource).

[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=75310553)](https://www.bountysource.com/issues/75310553-feature-add-proxy-support)

**Comment by Stebalien** [comment]
I've filed a meta-issue (https://github.com/ipfs/go-ipfs/issues/6430) to track anonymity in go-ipfs and explain our current position on the subject.

TL;DR: it's a hard problem we aren't going to solve any time soon.
